docs/src/guide/staking/index.md

```diff
@@ -0,0 +1,136 @@
+Let's look at a snippet from the perspective of a Deployer. 
+
+As the deployer, you want to set the reward rate, ASA ID of the token being stakes, the ASA ID of the rewarded token, and the initial supply of the reward token.
+
+``` rsh
+// Initiate the Deployer's local state
+Deployer.only(() => {
+    // this = address doing transaction
+    const deployerAddr = this;
+    // Declassify data and store in constants
+    const stakeToken = declassify(interact.setStakeToken());
+    const rewardRate = declassify(interact.setRewardRate());
+    const rewardToken = declassify(interact.setRewardToken());
+    const initialSupply = declassify(interact.setInitialSupply());
+});
+
+// Publish this information to global state
+Deployer.publish(
+    deployerAddr,
+    stakeToken,
+    rewardRate,
+    rewardToken,
+    initialSupply,
+);
+``` 
+
+The configurable contract is deployed. The next step is to enable users to interact with it. 
+In Reach, the solution is to add an API call so that when the user calls the function, they will pay `{!rsh} tokenAmt` of the `{!rsh} stakeToken` in the Deployer set. 
+
+``` rsh
+.api(User.stakeTokens,
+  (tokenAmt) => {
+      // validation that a user must stake more than 1 token
+      // they will receive the message on the frontend if it fails
+      assume(tokenAmt >= 1, 'Must stake at least 1 token');
+  },
+  (tokenAmt) => {
+      //looks unintuitive, but this means the user will
+      // pay 0 algos, and pay tokenAmt in rewardTokens
+      return [0, [tokenAmt, stakeToken]]
+  },
+  (tokenAmt, apiReturn) => {
+      require(tokenAmt >= 1, 'Must stake at least 1 token');
+
+      userLocalState[this] = {
+          stake: tokenAmt,
+          pendingRewards: 0,
+      }
+      // return to the API caller
+      apiReturn(true)
+  }
+)
+```
+
+## Unstaking
+
+Unstaking is similar in structure.
+
+``` rsh
+.api(User.unstakeAndClaim,
+  () => {
+      // No rewards left in the contract? Sry!
+      assume(balance(rewardToken) > 0, 'No rewards left')
+  },
+  () => {
+      // transfer nothing to the contract
+      return [0, [0, stakeToken]]
+  },
+  (apiReturn) => {
+      require(balance(rewardToken) > 0, 'No rewards left')
+      const userState = userLocalState[this];
+      // pay acrued rewards
+      transfer([[userState.pendingrewards, rewardToken]]).to(this);
+
+      userLocalState[this] = {
+          stake: 0,
+          pendingRewards: 0,
+      }
+      // return to the API caller
+      apiReturn(true)
+      // update global state
+      return [
+          rewardSupply - userState.pendingRewards
+      ]
+  }
+)
+```
+
+## Connect to View
+
+Now for the cool part: Use your contract to hook to your website.
+
+``` rsh
+// connect to the contract
+const application = account.contract(backend, APPLICATION_ID);
+// defines all User api calls (stakeTokens, unstaketokens)
+const userApi = contract.a.User;
+// this is a view; read the reach docs at docs.reach.sh
```

Comment:
This should be an actual link (by breaking up the code and/or relying on the automatic code linking)